src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -558,36 +568,42 @@ calculateFloatingPaymentClaimsFromSwapStream floatingRateCalculation s
 
       -- calculate all calculation period claims (using flat compounding) for a payment period
       createCalcPeriodClaim p =
-        mconcat $ calcFlatCompounding notional currency p s regularRateInclSpread
-          regularRateExclSpread rateFixingCalendar fixLeg paymentPeriodData []
+        mconcat $ calcFlatCompounding notional currency p s
+          floatingRateCalculation.floatingRateIndex floatingRateSpread rateFixingCalendar fixLeg
+          paymentPeriodData []
         where
           (notional, paymentPeriodData) = calculateNotionalAndFixings p s
             calculationScheduleStaticData fxRateId useAdjustedDatesForDcf periodicSchedule
 
       -- calculate simple interest rate claim (no calc period compounding) for a payment period
       -- using a non-compounded reference rate
-      createSimpleClaim regularRate p =
+      createSimpleClaim p =
         mconcat $ map createSimpleRate paymentPeriodData
         where
           (notional, paymentPeriodData) = calculateNotionalAndFixings p s
             calculationScheduleStaticData fxRateId useAdjustedDatesForDcf periodicSchedule
           createSimpleRate pp =
-            when (TimeGte pp.rateFixingDate) $ scale notional $
-            when (TimeGte pp.rateFixingDate) $ scale (rate * pp.dcf) $
+            when (TimeGte pp.rateFixingDate) $ scale notional $ scale (rate * pp.dcf) $
```

Comment:
I think we don't need the outer `When` node anymore when using `ObserveAt`. We can stick the observable directly after `when (TimeGte p.adjustedEndDate). Also, no need to use multiple `scale`

Reply:
The notional can have an FX rate observation in it, which I was planning on refactoring as part of a separate PR: https://github.com/digital-asset/daml-finance/issues/792

Reply:
I will try to remove the remaining `when (TimeGte pp.rateFixingDate)` nodes as part of that PR.